tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -0,0 +1,1034 @@
+/*
+ * (c) Copyright IBM Corp. 2015, 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#include "AixDwarfParser.hpp"
+
+
+/* Statics to create: */
+static unordered_map<double, Dwarf_Die> createdDies;
+static unordered_map<Dwarf_Off, Dwarf_Die> refMap;
+static str_vect _fileList;
+static Dwarf_CU_Context *_firstCU = NULL;
+static Dwarf_CU_Context *_currentCU = NULL;
+static Dwarf_Die _lastDie = NULL;
+
+static void deleteDie(Dwarf_Die die);
+static int parseDwarfDie(const string data);
+static int parseDwarfInfo(const char *line);
+static Dwarf_Half parseDeclarationLine (string data, Parsed_Declaration_Data *parsedDeclarationData );
+static void setError(Dwarf_Error *error, Dwarf_Half num);
+
+/* _startNewFile is a state counter */
+static int _startNewFile;
+static double _fileID;
+static const char *_errmsgs[] = {
+	"No error (0)",
+	"DW_DLE_ATTR_FORM_BAD (1)",
+	"DW_DLE_BADOFF (2) Invalid offset",
+	"DW_DLE_DIE_NULL (3)",
+	"DW_DLE_FNO (4) file not open",
+	"DW_DLE_IA (5) invalid argument",
+	"DW_DLE_IOF (6) I/O failure",
+	"DW_DLE_MAF (7) memory allocation failure",
+	"DW_DLE_NOB (8) not an object file or dSYM bundle not found",
+	"DW_DLE_VMM (9) dwarf format/library version mismatch",
+};
+
+/* temporary debugging methods */
+void printDie (Dwarf_Die* die, int spaces) {
```

Comment:
Functions main(), printCU(), printDie(), etc should not be in here. If you want *minimal* debug output, use the "DEBUGPRINTF" macro (as used in DwarfScanner.cpp) and not cout. I suggest using it during normal operation, such as printing out a CU or Die name as you create it, and not in a separate printSomething() function.